### PR TITLE
New Port Kaima

### PR DESCRIPTION
**Pull request by Tlaggypyro**
# New Port for { Kaima}

## Game Information
- **Title**: {Kaima}
- **URL**: { https://nomnomnami.itch.io/kaima }

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/kaima/kaima.sh
Comment (cdeletre) [inline ports/kaima/kaima.sh:1]: The name of the `.sh` script should start with uppercase letter.

File: ports/kaima/Kaima.sh
Comment (kloptops) [inline ports/kaima/Kaima.sh:1]: Not sure what has happened here, but the file is NULL padded at the end?

File: ports/kaima/README.md
Comment (kloptops) [inline ports/kaima/README.md:1]: And this file is null padded.

File: ports/kaima/gameinfo.xml
Comment (kloptops) [inline ports/kaima/gameinfo.xml:1]: And this file...

File: ports/kaima/kaima/conf/mkxp.conf
Comment (kloptops) [inline ports/kaima/kaima/conf/mkxp.conf:1]: And this file....

File: ports/kaima/kaima/kaima.gptk
Comment (kloptops) [inline ports/kaima/kaima/kaima.gptk:1]: And again.

File: ports/kaima/port.json
Comment (kloptops) [inline ports/kaima/port.json:1]: this too.

**Review by kloptops — CHANGES_REQUESTED** [review]
Lots of files are NULL padded. Not sure what has happened here. 🐢 🤔

Screenshot 2024-08-21 at 7 42 54 PM

File: ports/kaima/Kaima.sh
Comment (Tlaggypyro) [reply ports/kaima/Kaima.sh:1]: How do I fix that?

Comment (cdeletre) [reply ports/kaima/Kaima.sh:1]: With Linux you can run this command
`tr < README.md -d '\000' > README-clean.md`

Comment (cdeletre) [reply ports/kaima/Kaima.sh:1]: With Windows you can use this trick https://stackoverflow.com/questions/19332949/removing-nul-characters which works in Notepad++. It should work in other text editors such as Visual Studio Code (just check how to use regular expression)

Comment (kloptops) [reply ports/kaima/Kaima.sh:1]: > How do I fix that?

I am not sure how you even managed it?

Was it a bug in uploading to github?

**Comment by Tlaggypyro** [comment]
I reuploaded everything did that fix it by chance

File: ports/kaima/LICENSE_falcon-mkxp
Comment (JeodC) [inline ports/kaima/LICENSE_falcon-mkxp:1]: This file needs to go in the portfolder.

File: ports/kaima/Kaima.sh
Comment (Cebion) [inline ports/kaima/Kaima.sh:1]: Nope, still broken

image

Comment (Cebion) [reply ports/kaima/Kaima.sh:1]: File is in ANSI and has NUL.
Not sure how you edited / saved these files.
Best to copy your changes into new files.
On a PC.

**Review by Cebion — CHANGES_REQUESTED** [review]

**Pull request closed without merging**